server/app/routes/cart/index.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const router = require('express').Router();
+const mongoose = require('mongoose');
+const User = mongoose.model('User');
+const Promise = require('bluebird');
+
+
+router.get('/', function (req, res, next) {
+	if(req.user) {
+		User.findById(req.user._id)
+		.populate('cart.product')
+		.then(function (user) {
+			res.status(200).json(user.cart);
+		})
+	}
+	else {
+		//THIS DOESNT WORKAHDGFIHASDFUHOASDFOHa!!!!!!??????
+		if(!req.session.cart) req.session.cart = []; //initialize a session cart if needed
+		console.log("IM CURRENTLY HERE");
+		Promise.all(req.session.cart, function (item) {
```

Comment:
I think something like the following will work:
`var productsPromises = [];
req.session.cart.forEach(function (item) {
    productsPromises.push(Product.findById(item.product).exec()) //don't forget to include mongoose Product model
})
Promise.all(productsPromises, cb)
`